### ISSUE TITLE
install error

### ISSUE THREAD

**Issue by zhangda89** [ErikDubbelboer/node-sleep#38]
```
make: *** [Release/obj.target/node_sleep/sleep.o] 错误 1
make: Leaving directory `/data/mocker.qeeniao.com/node_modules/sleep/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/zhangdapeng/.nvm/versions/node/v0.12.4/lib/node_modules/cnpm/node_modules/pangyp/lib/build.js:271:23)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
gyp ERR! System Linux 2.6.32-504.8.1.el6.x86_64
gyp ERR! command "node" "/home/zhangdapeng/.nvm/versions/node/v0.12.4/lib/node_modules/cnpm/node_modules/pangyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /data/mocker.qeeniao.com/node_modules/sleep
gyp ERR! node -v v0.12.4
gyp ERR! pangyp -v v2.1.0
gyp ERR! not ok 
npm ERR! Linux 2.6.32-504.8.1.el6.x86_64
npm ERR! argv "node" "/home/zhangdapeng/.nvm/versions/node/v0.12.4/lib/node_modules/cnpm/node_modules/.bin/npm" "--userconfig=/home/zhangdapeng/.cnpmrc" "--disturl=https://npm.taobao.org/mirrors/node" "--cache=/home/zhangdapeng/.cnpm" "--registry=https://registry.npm.taobao.org" "--node-gyp=/home/zhangdapeng/.nvm/versions/node/v0.12.4/lib/node_modules/cnpm/node_modules/pangyp/bin/node-gyp.js" "install" "sleep" "--save"
npm ERR! node v0.12.4
npm ERR! npm  v2.7.6
npm ERR! code ELIFECYCLE
```

**Comment by ErikDubbelboer** [comment]
Which version of node-sleep are you trying to install? Can you please try `V=1 npm --loglevel verbose install sleep` and paste the output?

**Comment by menicosia** [comment]
Yep, I'm having the same problem.

Here's a gist of the output of `V=1 npm --loglevel verbose install sleep`
https://gist.github.com/menicosia/6ce6d8db8c28320a98e7

Here's the output of `npm ls`
```
trivial-js@0.0.1 /Users/marco/workspace/trivial-js
├── nan@2.0.5
├── UNMET DEPENDENCY sleep@2.0.0
└── strftime@0.8.2

npm ERR! missing: sleep@2.0.0, required by trivial-js@0.0.1
```
It suddenly stopped working. I can see from the history of my packages.json file that sleep@2.0.0 was working for me back in mid-June. But now in August, it's not working. I'm sure I must be doing something differently, or upgraded something that's not compatible, but I don't know how to figure out what it is.

-- Marco N.

**Comment by menicosia** [comment]
I eventually managed to get things back to sane again by requiring nan 1.x and sleep 2.0.0.

-- Marco N.

**Comment by ErikDubbelboer** [comment]
The issue is that sleep 2.0.0 doesn't depend on a specific version of nan. So when nan published their breaking changes sleep 2.0.0 broken.

Updating sleep to 2.0.1 or 3.0.0 fixes this issue. sleep 2.0.0 will for always be broken seeing as it's impossible to overwrite an already existing version on npm.

**Comment by menicosia** [comment]
@ErikDubbelboer, okay! Thanks for explaining what's going on.

-- Marco N.
